pkg/eks/components/CruEKS.vue

```diff
@@ -0,0 +1,766 @@
+          }
+
+          return !!this.nodeGroups.find((group: EKSNodeGroup) => !group.instanceType && !group.requestSpotInstances) ? this.t('validation.required', { key: this.t('eks.nodeGroups.instanceType.label') }) : null;
+        },
+        desiredSize: (size) => {
+          const msg = this.t('eks.errors.greaterThanZero', { key: this.t('eks.nodeGroups.desiredSize.label') });
+
+          if (size !== undefined) {
+            return size > 0 ? null : msg;
+          }
+
+          return !!this.nodeGroups.find((group) => !group.desiredSize || group.desiredSize <= 0) ? msg : null;
+        },
+        subnets: (val) => {
+          const subnets = val || this.config.subnets;
+
+          return subnets && subnets.length === 1 ? this.t('eks.errors.minimumSubnets') : undefined;
+        },
+        publicPrivateAccess: () => {
+          const { publicAccess, privateAccess } = this.config;
+
+          return publicAccess || privateAccess ? undefined : this.t('eks.errors.publicOrPrivate');
+        },
+
+      };
+    },
+
+    // upstreamSpec will be null if the user created a cluster with some invalid options such that it ultimately fails to create anything in aks
+    // this allows them to go back and correct their mistakes without re-making the whole cluster
+    isNewOrUnprovisioned() {
+      return this.mode === _CREATE || !this.normanCluster?.eksStatus?.upstreamSpec;
+    },
+
+    isEdit() {
+      return this.mode === _CREATE || this.mode === _EDIT;
+    },
+
+    doneRoute() {
+      return this.value?.listLocation?.name;
+    },
+
+    hasCredential() {
+      return !!this.config?.amazonCredentialSecret;
+    },
+
+    clusterId(): String | null {
+      return this.value?.id || null;
+    },
+
+    canManageMembers(): Boolean {
+      return canViewClusterMembershipEditor(this.$store);
+    },
+
+    CREATE(): string {
+      return _CREATE;
+    },
+
+    VIEW(): string {
+      return _VIEW;
+    },
+
+    groupedInstanceTypes() {
+      const out = {} as any;
+
+      this.instanceTypes.forEach((type:any) => {
+        if (out[type.groupLabel]) {
+          out[type.groupLabel].push(type);
+        } else {
+          out[type.groupLabel] = [type];
+        }
+      });
+
+      return out;
+    },
+
+    instanceTypeOptions() {
+      const out = [] as any[];
+
+      Object.keys(this.groupedInstanceTypes).forEach((groupLabel: string) => {
+        const instances = this.groupedInstanceTypes[groupLabel];
+        const groupOption = { label: groupLabel, kind: 'group' };
+        const instanceTypeOptions = instances.map((instance:any) => {
+          return {
+            value: instance.apiName,
+            label: instance.label,
+            group: instance.groupLabel
+          };
+        });
+
+        out.push(groupOption);
+        out.push(...instanceTypeOptions);
+      });
+
+      return out;
+    },
+
+    spotInstanceTypeOptions() {
+      const out = [] as any[];
```

Comment:
same as above, so probably should be an official type . guessing they relate to a component as well, so should live near there. 

const out: {label: string, value?: string, group?: string, kind?: string}[] = [];